<a href="https://colab.research.google.com/github/DLuisMontufar3/1/blob/main/Si5.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**Ejercicio 5** 05 Auditoría de Desempeño para Negocios Digitales:

En este ejercicio vamos a analizar un dataset de transacciones y determinar si la estrategia de M-Commerce (App Móvil) es más rentable que el E-Commerce tradicional (Web Desktop).

Que canal genera mas ingresos?

In [14]:
import pandas as pd
import numpy as np

# Semilla para reproducibilidad
np.random.seed(42)

# Crear datos simulados
data = {
    'ID_Transaccion': range(1001, 1051),
    'Canal': np.random.choice(['Web Desktop', 'App Móvil'], 50, p=[0.4, 0.6]),
    'Metodo_Pago': np.random.choice(['Tarjeta Crédito', 'PayPal', 'Transferencia'], 50),
    'Total_Venta': np.random.randint(20, 500, 50), # Ventas entre $20 y $500
    'Categoria': np.random.choice(['Electrónica', 'Moda', 'Hogar'], 50)
}

df = pd.DataFrame(data)

df

ID_Transaccion        Canal      Metodo_Pago  Total_Venta    Categoria
0             1001  Web Desktop    Transferencia          235         Moda
1             1002    App Móvil           PayPal          312         Moda
2             1003    App Móvil           PayPal          118  Electrónica
3             1004    App Móvil           PayPal          191         Moda
4             1005  Web Desktop           PayPal          379  Electrónica
5             1006  Web Desktop           PayPal          233  Electrónica
6             1007  Web Desktop           PayPal          494         Moda
7             1008    App Móvil    Transferencia           54         Moda
8             1009    App Móvil    Transferencia          468         Moda
9             1010    App Móvil           PayPal          246        Hogar
10            1011  Web Desktop    Transferencia          120         Moda
11            1012    App Móvil  Tarjeta Crédito          450        Hogar
12            1013    App Móvil           PayPal          481  Electrónica
13            1014  Web Desktop  Tarjeta Crédito          150        Hogar
14            1015  Web Desktop  Tarjeta Crédito          276         Moda
15            1016  Web Desktop           PayPal           24  Electrónica
16            1017  Web Desktop    Transferencia          237  Electrónica
17            1018    App Móvil  Tarjeta Crédito          274  Electrónica
18            1019    App Móvil           PayPal          417        Hogar
19            1020  Web Desktop  Tarjeta Crédito          378         Moda
20            1021    App Móvil  Tarjeta Crédito          302  Electrónica
21            1022  Web Desktop  Tarjeta Crédito          412  Electrónica
22            1023  Web Desktop  Tarjeta Crédito          226  Electrónica
23            1024  Web Desktop    Transferencia           34        Hogar
24            1025    App Móvil  Tarjeta Crédito          365        Hogar
25            1026    App Móvil  Tarjeta Crédito           61         Moda
26            1027  Web Desktop  Tarjeta Crédito          399        Hogar
27            1028    App Móvil    Transferencia          480  Electrónica
28            1029    App Móvil  Tarjeta Crédito          198         Moda
29            1030  Web Desktop  Tarjeta Crédito           82  Electrónica
30            1031    App Móvil    Transferencia          371  Electrónica
31            1032  Web Desktop    Transferencia          250        Hogar
32            1033  Web Desktop    Transferencia          260         Moda
33            1034    App Móvil  Tarjeta Crédito           71        Hogar
34            1035    App Móvil    Transferencia          115        Hogar
35            1036    App Móvil    Transferencia          407         Moda
36            1037  Web Desktop  Tarjeta Crédito          241  Electrónica
37            1038  Web Desktop    Transferencia          426  Electrónica
38            1039    App Móvil  Tarjeta Crédito          250         Moda
39            1040    App Móvil           PayPal          256  Electrónica
40            1041  Web Desktop    Transferencia          162         Moda
41            1042    App Móvil           PayPal          190         Moda
42            1043  Web Desktop  Tarjeta Crédito           48        Hogar
43            1044    App Móvil    Transferencia           55         Moda
44            1045  Web Desktop  Tarjeta Crédito           32        Hogar
45            1046    App Móvil           PayPal          179  Electrónica
46            1047  Web Desktop  Tarjeta Crédito          346  Electrónica
47            1048    App Móvil    Transferencia          206  Electrónica
48            1049    App Móvil    Transferencia          262  Electrónica
49            1050  Web Desktop           PayPal          105        Hogar

In [15]:
#Kpis
#1. Volumen de ventas por canal.
#2. Ticket promedio (promedio de  ingresos).
#3. Preferencia de pagos (En cada comerce cual metodo de pago se uso mas).
#4. Categoria por canal (Que se compra mas en cada uno de los comerce).
#"2"
# Umbral de cuotas.

In [16]:
df['Canal'].value_counts()

Canal
App Móvil      26
Web Desktop    24
Name: count, dtype: int64

El canal más ocupado es el es M-commerce (App móvil)

In [17]:
ingreso_totalcanal = df.groupby('Canal')['Total_Venta'].sum()
ingreso_totalcanal

Canal
App Móvil      6779
Web Desktop    5549
Name: Total_Venta, dtype: int64

In [18]:
diferencia = ingreso_totalcanal['App Móvil'] - ingreso_totalcanal['Web Desktop']
diferencia

np.int64(1230)

En M-commerce (App móvil) genero mas ingresos totales con una diferencia de 1230

In [19]:
ingreso_total = df['Total_Venta'].sum()
ingreso_total

np.int64(12328)

In [20]:
ingreso_prom = df.groupby('Canal')['Total_Venta'].mean()
ingreso_prom

Canal
App Móvil      260.730769
Web Desktop    231.208333
Name: Total_Venta, dtype: float64

El canal que genero mas ingresos es M-commerce (App movil)

In [21]:
df.groupby('Canal')['Categoria'].value_counts()


Canal        Categoria  
App Móvil    Electrónica    10
             Moda           10
             Hogar           6
Web Desktop  Electrónica    10
             Hogar           7
             Moda            7
Name: count, dtype: int64

En M-commerce se utilizó  mas para compras de Electronica y Moda.
En E-commerce se utlizó  mas para Electrónica

In [22]:
df.groupby('Canal')['Metodo_Pago'].value_counts()

Canal        Metodo_Pago    
App Móvil    PayPal              9
             Transferencia       9
             Tarjeta Crédito     8
Web Desktop  Tarjeta Crédito    11
             Transferencia       8
             PayPal              5
Name: count, dtype: int64

En M-commerce se utilizó  mas pagos en Paypal y transferencia.
En E-commerce se utlizó  mas pagos con Tarjeta de crédito

In [23]:
url = "https://raw.githubusercontent.com/DLuisMontufar3/1/refs/heads/main/datos/olist_order_payments_dataset.csv"
df = pd.read_csv(url)

df.head()


order_id  payment_sequential payment_type  \
0  b81ef226f3fe1789b1e8b2acac839d17                   1  credit_card   
1  a9810da82917af2d9aefd1278f1dcfa0                   1  credit_card   
2  25e8ea4e93396b6fa0d3dd708e76c1bd                   1  credit_card   
3  ba78997921bbcdc1373bb41e913ab953                   1  credit_card   
4  42fdf880ba16b47b59251dd489d4441a                   1  credit_card   

   payment_installments  payment_value  
0                     8          99.33  
1                     1          24.39  
2                     1          65.71  
3                     8         107.78  
4                     2         128.45

In [24]:
df['payment_type'].value_counts()

payment_type
credit_card    76795
boleto         19784
voucher         5775
debit_card      1529
not_defined        3
Name: count, dtype: int64

In [25]:
df['payment_installments'].value_counts()

payment_installments
1     52546
2     12413
3     10461
4      7098
10     5328
5      5239
8      4268
6      3920
7      1626
9       644
12      133
15       74
18       27
11       23
24       18
20       17
13       16
14       15
17        8
16        5
21        3
0         2
22        1
23        1
Name: count, dtype: int64

In [26]:
max_value = df['payment_value'].max()
min_value = df['payment_value'].min()

print("Pago alto:", max_value)
print("Pago bajo:", min_value)


Pago alto: 13664.08
Pago bajo: 0.0


In [27]:
total_transacciones = len(df)
print("Total de transacciones:", total_transacciones)

Total de transacciones: 103886


In [28]:
ingreso_prom = df.groupby('payment_installments')['payment_value'].mean()
ingreso_prom

payment_installments
0      94.315000
1     112.420229
2     127.228150
3     142.539317
4     163.976840
5     183.465222
6     209.849952
7     187.673672
8     307.737427
9     203.440870
10    415.085837
11    124.932174
12    321.678496
13    150.462500
14    167.962667
15    445.553108
16    292.694000
17    174.602500
18    486.483333
20    615.801765
21    243.700000
22    228.710000
23    236.480000
24    610.048889
Name: payment_value, dtype: float64